File: src/frontend/src/models/sensor.ts
import { ConnectionState } from "./general";

export enum SensorType {
  TEMPERATURE = "TEMPERATURE",
  HUMIDITY = "HUMIDITY",
  LIGHT = "LIGHT",
}

export type SensorRef = {
  id: string;
  type: SensorType;
  key?: string;
};

export type Sensor = SensorRef & {
  name: string;
  description: string;
  currentValue: number;
  minValue: number;
  maxValue: number;
  unit: string;
};

export type SensorData = {
  sensorKey: string;
  connectionState: ConnectionState;
  currentValue: number;
  min: number;
  max: number;
  unit: string;
  lastUpdate: Date;
};

File: src/backend/SmartGarden.EntityFramework/Models/Sensor.cs
using SmartGarden.EntityFramework.Enums;

namespace SmartGarden.EntityFramework.Models;

public class Sensor : BaseEntityWithOrder
{
    public string Name { get; set; }
    public string Description { get; set; }
    public double Min { get; set; }
    public double Max { get; set; }
    public string Unit { get; set; }
    public SensorType Type { get; set; }

    // TODO: Add Properties to connect actual Sensor to app
}